nuxeo-webdav/src/main/java/org/nuxeo/ecm/webdav/resource/ExistingResource.java

```diff
@@ -189,16 +234,23 @@ private Response copyOrMove(String method, @HeaderParam("Destination") String de
             return Response.status(409).build();
         }
 
-        if ("COPY".equals(method)) {
-            DocumentModel destDoc = backend.copyItem(doc, destParentRef);
-            backend.renameItem(destDoc, getNameFromPath(destPath));
-        } else if ("MOVE".equals(method)) {
-            if (backend.isRename(doc.getPathAsString(), destPath)) {
-                backend.renameItem(doc, getNameFromPath(destPath));
-            } else {
-                backend.moveItem(doc, destParentRef);
-            }
+        switch (method) {
+            case "COPY":
+                DocumentModel destDoc = backend.copyItem(doc, destParentRef);
+                backend.copyItem(destDoc, new PathRef(getParentPath(destPath)));
+                break;
+            case "MOVE":
+                String docPath = doc.getPathAsString();
+                DocumentModel newDoc = backend.moveItem(doc, destParentRef, getNameFromPath(destPath));
+
+                if (destPath.endsWith(".tmp")) {
+                    newDoc.setPropertyValue("dc:source", docPath);
+                    backend.getSession().saveDocument(newDoc);
```

Comment:
What if the input blob submitted by user finished by `.tmp` but don't reference the right blob in the logic ?